build.rs

```diff
@@ -1,18 +1,41 @@
 use std::env;
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
 
 use libbpf_cargo::SkeletonBuilder;
 
 const SRC: &str = "src/bpf/hidrecord.bpf.c";
 const SRC_TRACING: &str = "src/bpf/hidrecord_tracing.bpf.c";
 
+// from https://stackoverflow.com/questions/37498864/finding-executable-in-path-with-rust
+fn which<P>(exe_name: P) -> Option<PathBuf>
+where
+    P: AsRef<Path>,
+{
+    env::var_os("PATH").and_then(|paths| {
+        env::split_paths(&paths)
+            .filter_map(|dir| {
+                let full_path = dir.join(&exe_name);
+                if full_path.is_file() {
+                    Some(full_path)
+                } else {
+                    None
+                }
+            })
+            .next()
+    })
+}
+
 fn main() {
-    let mut out = PathBuf::from(
-        env::var_os("OUT_DIR").expect("expected OUT_DIR to be set by cargo but it's empty"),
-    );
-    let mut out_tracing = out.clone();
-    out.push("hidrecord.skel.rs");
-    out_tracing.push("hidrecord_tracing.skel.rs");
+    if which("clang").is_none() {
+        println!(
+            "cargo:warning='clang' executable not found, using provided bpf compilation outputs"
+        );
+        println!("cargo:warning= changes in 'src/bpf/*' will not be taken into account");
+        return;
+    }
```

Comment:
*handwavy*: instead of reimplementing `which` wouldn't it better to try to compile and detect a compilation failure? Not sure the error gives us enough detail for this thoug...

Reply:
done in #34, please feel free to cherry-pick and squash